app/main.py

```diff
@@ -1,2 +1,103 @@
+from __future__ import annotations
+from typing import Any, Iterator
+
+
+class Node:
```

Comment:
This class better be defined inside of a Dictionary since it is a part of its implementation and will not be used outside of it